test/color.py

```diff
@@ -0,0 +1,47 @@
+"""Test Color Sensor functionality"""
+import time
+import unittest
+
+from buildhat import ColorSensor
+
+
+class TestColor(unittest.TestCase):
+    """Test color sensor functions"""
+
+    def test_color(self):
+        """Test color sensor interval"""
+        color = ColorSensor('A')
+        color.avg_reads = 1
+        color.interval = 10
+        count = 1000
+        expecteddur = count * color.interval * 1e-3
+
+        start = time.time()
+        for _ in range(count):
+            color.get_ambient_light()
+        end = time.time()
+        diff = abs((end - start) - expecteddur)
+        self.assertLess(diff, 0.25)
+
+        start = time.time()
+        for _ in range(count):
+            color.get_color_rgbi()
+        end = time.time()
+        diff = abs((end - start) - expecteddur)
+        self.assertLess(diff, 0.25)
```

Comment:
Would these make more sense as separate tests with a before block for the setup?

Reply:
I can see what you mean but they're both just testing the setting of the interval works correctly.  Would it be ok if I left as-is and called the function something like ```test_color_interval```

Reply:
Okay then, let's go with that

Reply:
Just done